android/src/org/kiwix/kiwixmobile/LibraryFragment.java

```diff
@@ -103,6 +105,33 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
         noNetworkConnection();
       }
 
+      this.getContext().registerReceiver(new BroadcastReceiver() {
+        @Override
+        public void onReceive(Context context, Intent intent) {
+
+          permissionButton.setVisibility(View.GONE);
+          permissionText.setVisibility(View.GONE);
+          progressBar.setVisibility(View.INVISIBLE);
+          progressBarLayout.setVisibility(View.GONE);
+          progressBarMessage.setVisibility(View.GONE);
+
+          NetworkInfo network = conMan.getActiveNetworkInfo();
+
+          if (network != null && network.isConnected()) {
+            if (books == null) {
+              if (isWiFi()) {
+                getLibraryData();
+              } else {
+                displayNetworkConfirmation();
```

Comment:
books == null is not telling the truth here for me. Its null even when there are books. Not quite sure why. If you can't figure out (or reproduce) ask and I can look deeper into it.